packages/slate/src/create-editor.ts

```diff
@@ -139,20 +142,31 @@ export const createEditor = (): Editor => {
       const { selection } = editor
 
       if (selection) {
-        return Node.fragment(editor, selection)
+        const fragment = Node.fragment(editor, selection)
+        return fragment
       }
+
       return []
     },
 
     insertBreak: () => {
       Transforms.splitNodes(editor, { always: true })
     },
 
-    insertFragment: (fragment: Node[]) => {
+    // @ts-ignore
+    insertFragment: (
+      fragment: Array<ElementOf<Editor<V>> | TextOf<Editor<V>>>
+    ) => {
       Transforms.insertFragment(editor, fragment)
     },
 
-    insertNode: (node: Node) => {
+    // @ts-ignore
+    insertNode: (
+      node:
+        | ElementOf<Editor<V>>
+        | TextOf<Editor<V>>
+        | Array<ElementOf<Editor<V>> | TextOf<Editor<V>>>
```

Comment:
To tackle ts error, we could set:
```suggestion
       VElement<V> | VText<V> | Array<VElement<V> | VText<V>>
```
where
```tsx
export type VElement<V extends Value> = ElementOf<V[0]>;
export type VText<V extends Value> = TextOf<V[0]>;
```
that way, we don't use `Editor<V>` but just `V[0]`